src/nevermined/api/ComputeApi.ts

```diff
@@ -0,0 +1,152 @@
+     * @returns The metadata of the asset created (DDO)
+     *
+     * @returns {@link DDO}
+     */
+     public createRefactored(
+        assetAttributes: AssetAttributes,
+        publisher: Account,
+        publishMetadata: PublishMetadata = PublishMetadata.OnlyMetadataAPI,
+        txParams?: TxParameters
+    ): SubscribablePromise<CreateProgressStep, DDO> {
+        return this.registerNeverminedAsset(
+            assetAttributes,
+            publisher,
+            publishMetadata,
+            undefined,
+            txParams
+        )
+    }
+
+
+    /**
+     * Given a DID, updates the metadata associated to the asset. It also can upload this metadata to a remote decentralized stored depending on the `publishMetadata` parameter.
+     * 
+     * @example
+     * ```ts
+     * const ddoUpdated = await nevermined.compute.update(
+     *      ddo.shortId(), 
+     *      updatedMetadata, 
+     *      publisher, 
+     *      PublishMetadata.IPFS
+     * )
+     * ```
+     * 
+     * @param did - Decentralized ID representing the unique id of an asset in a Nevermined network.
+     * @param metadata - Metadata describing the asset
+     * @param publisher - Account of the user updating the metadata
+     * @param publishMetadata - It allows to specify where to store the metadata  
+     * @param txParams - Optional transaction parameters
+     * @returns {@link DDO} The DDO updated
+     */    
+    public update(
+        did: string,
+        metadata: MetaData,
+        publisher: Account,
+        publishMetadata: PublishMetadata = PublishMetadata.OnlyMetadataAPI,
+        txParams?: TxParameters
+    ): SubscribablePromise<UpdateProgressStep, DDO> {
+        return this.updateAsset(did, metadata, publisher, publishMetadata, txParams)
+    }
+
+    /**
+     * Start the purchase/order of a compute service. Starts by signing the service agreement
+     * then sends the request to the publisher via the service endpoint (Node http service).
+     * @param did - Unique identifier of the asset to order
+     * @param consumerAccount - The account of the user ordering the asset
+     * @param txParams - Optional transaction parameters
+     * @returns The agreement ID identifying the order     
+     */
+     public order(
+        did: string,
+        consumerAccount: Account,
+        txParams?: TxParameters
+    ): SubscribablePromise<OrderProgressStep, string> {
+        return this.orderAsset(did, 'compute', consumerAccount, txParams)
+    }
+
+    /**
+     * It trigers the execution of a compute job
+     * @param agreementId - The unique identifier of the order placed for a service
+     * @param workflowDid - The unique identifier of the Asset representing the workflow
+     * @param consumerAccount - The account of the user trigerring the computation
+     * @returns If the execution is correct it returns the response given by the Nevermined Node 
+    */
+    public async execute(
+        agreementId: string,
```

Comment:
not sure we need to keep using the agreementId

Reply:
same as access

Reply:
for multiple orders around the same asset it's necessary to distinguish between them using the `agreementId`